axlearn/common/attention.py

```diff
@@ -2984,13 +3003,16 @@ def _forward_for_mode(
                 None,
                 self.self_attention(
                     target=data,
+                    segment_ids=segment_ids,
                     source=self_attention_kv_state,
                     attention_logit_biases=self_attention_logit_biases,
                     return_aux=self_attention_return_aux,
                 ),
             )
         elif mode == ForwardMode.INIT_STATES:
             assert cached_states is not None
+            if segment_ids is not None:
+                raise ValueError("segment_ids is not supported in INIT_STATES.")
```

Comment:
In theory we could prefill on multiple segments -- maybe `NotImplementedError` is more suitable, but feel free to leave as-is.

Reply:
Makes sense.